File: Arrays & Hashing/1404. Number of Steps to Reduce a Number in Binary Representation to One/1404.py
class Solution:
    def numSteps(self, s: str) -> int:
        s=list(s)
        res=0
        while (len(s)>1):
            if (s[-1]=='0'): s.pop()
            else:
                i=len(s)-1
                while (i>=0 and s[i]=='1'):
                    s[i]='0'
                    i-=1
                if (i>=0): s[i]='1'
                else: s.insert(0,'1')
            res+=1
        return res

File: Arrays & Hashing/1404. Number of Steps to Reduce a Number in Binary Representation to One/1404.cpp
#pragma GCC optimize("O3","Oz","unroll-loops","inline-functions","fast-math","march=native","lto","omit-frame-pointer")
class Solution {
public:
    int numSteps(string s) {
        ios_base::sync_with_stdio(0);
        cin.tie(0);
        cout.tie(0);

        vector<char>sList(s.begin(),s.end());
        int res=0;

        while (sList.size()>1) {
            if (sList[sList.size()-1]=='0') sList.pop_back();
            else {
                int i=sList.size()-1;
                while (i>=0 && sList[i]=='1') {
                    sList[i]='0';
                    i--;
                }
                if (i>=0) sList[i]='1';
                else sList.insert(sList.begin(),'1');
            }
            res++;
        }
        return res;
    }
};